src/main/java/uk/gov/register/db/EntryItemDAO.java

```diff
@@ -0,0 +1,12 @@
+package uk.gov.register.db;
+
+import org.skife.jdbi.v2.sqlobject.SqlBatch;
+import org.skife.jdbi.v2.sqlobject.customizers.BatchChunkSize;
+import uk.gov.register.core.Entry;
+import uk.gov.register.store.postgres.BindEntry;
+
+public interface EntryItemDAO {
+    @SqlBatch("insert into entry_item(entry_number, sha256hex) values (:entry_number, :sha256hex)")
+    @BatchChunkSize(1000)
+    void insertInBatch(@BindEntry Iterable<Entry> entries);
```

Comment:
This will not insert multiple rows if the entry has more than one item-hash. `BindEntry` still calls `getSha256hex` instead of the new `getItemHashes`. Remember also that an entry could have an empty list of hashes, in which case you would not insert anything in this table.